cloud-azure/src/main/java/com/sequenceiq/cloudbreak/cloud/azure/client/AzureClientFactory.java

```diff
@@ -0,0 +1,164 @@
+
+    private final AzureHttpClientConfigurer azureHttpClientConfigurer;
+
+    public AzureClientFactory(
+            CloudContext cloudContext,
+            AzureCredentialView credentialView,
+            ExecutorService mdcCopyingThreadPoolExecutor,
+            AzureHttpClientConfigurer azureHttpClientConfigurer) {
+        this.credentialView = credentialView;
+        azureEnvironment = getAzureEnvironment(Optional.ofNullable(cloudContext)
+                .map(CloudContext::getLocation)
+                .map(Location::getRegion));
+        azureProfile = new AzureProfile(credentialView.getTenantId(), credentialView.getSubscriptionId(), azureEnvironment);
+        this.mdcCopyingThreadPoolExecutor = mdcCopyingThreadPoolExecutor;
+        this.azureHttpClientConfigurer = azureHttpClientConfigurer;
+        azureCredential = getAzureCredentials();
+    }
+
+    public AzureEnvironment getAzureEnvironment() {
+        return azureEnvironment;
+    }
+
+    public AzureResourceManager getAzureResourceManager() {
+        return azureHttpClientConfigurer.configureDefault(AzureResourceManager.configure())
+                .authenticate(azureCredential, azureProfile)
+                .withDefaultSubscription();
+    }
+
+    public AzureResourceManager getAzureResourceManager(String subscriptionId) {
+        return azureHttpClientConfigurer.configureDefault(AzureResourceManager.configure())
+                .authenticate(azureCredential, new AzureProfile(credentialView.getTenantId(), subscriptionId, azureEnvironment))
+                .withDefaultSubscription();
+    }
+
+    public <T extends HttpTrait<T>> T configureDefault(T configurable) {
+        return azureHttpClientConfigurer.configureDefault(configurable);
+    }
+
+    public PrivateDnsZoneManager getPrivateDnsManager() {
+        return azureHttpClientConfigurer.configureDefault(PrivateDnsZoneManager.configure()).authenticate(azureCredential, azureProfile);
+    }
+
+    public PrivateDnsZoneManager getPrivateDnsManagerWithAnotherSubscription(String subscriptionId) {
+        return azureHttpClientConfigurer.configureDefault(PrivateDnsZoneManager.configure())
+                .authenticate(azureCredential, new AzureProfile(credentialView.getTenantId(), subscriptionId, azureEnvironment));
+    }
+
+    public ComputeManager getComputeManager() {
+        return azureHttpClientConfigurer.configureDefault(ComputeManager.configure()).authenticate(azureCredential, azureProfile);
+    }
+
+    public MarketplaceOrderingManager getMarketplaceOrderingManager() {
+        return azureHttpClientConfigurer.configureDefault(MarketplaceOrderingManager.configure()).authenticate(azureCredential, azureProfile);
+    }
+
+    public Optional<String> getAccessToken() {
+        return Optional.of(azureCredential.getTokenSync(new TokenRequestContext().addScopes("https://management.azure.com//.default")).getToken());
+    }
+
+    @VisibleForTesting
```

Comment:
is this necessary? i think it would be better to test private methods through public methods and not use test related code in production code

Reply:
Yes you are right I removed it.

Reply:
thanks!